data/mods/Xedra_Evolved/eocs/spell_learning_eoc.json

```diff
@@ -55,11 +55,17 @@
   {
     "type": "effect_on_condition",
     "id": "EOC_XE_QUEUE_LEVELING",
-    "recurrence": [ "1 h", "3 h" ],
-    "condition": { "u_has_effect": "sleep" },
+    "eoc_type": "EVENT",
+    "required_event": "character_wakes_up",
```

Comment:
Thematically it seems more appropriate to gain dreaming spells _while dreaming_ (approximated here as _being asleep_) instead of after waking up.

Reply:
it is technically right before you wake up, and it's mostly for better UX - you can go sleep, go away, then return and find that you didn't actually slept because the game poped up the selector

Reply:
This is a good fix. Thanks Guardian. Anything that improves the mechanics around sleeping to be less painful is an improvement.

Reply:
Oh right I remember that can happen (and can be quite inconvenient when things are slow and you used the 'sleeping' time to do something else outside the game).